tools/topology/m4/utils.m4

```diff
@@ -57,6 +57,15 @@ define(`COMP_FORMAT_VALUE',
 	$1, `s32le', `0x02',
 	)')
 
+dnl CAPABILITY_FORMAT_NAME(BUF_FMT)
+define(`CAPABILITY_FORMAT_NAME',
+`ifelse(
+	$1, `s16le', `S16_LE',
+	$1, `s24le', `S24_LE,S16_LE',
+	$1, `s32le', `S32_LE,S24_LE,S16_LE',
+	$1, `float', `FLOAT_LE',
```

Comment:
What's the rational here, do you want to mask the format names so one name can be used to describe more than one format ? i.e. ```s32s24le``` would mean 32bit and 24 bit but NOT 16bit.

Reply:
No, I am not gonna change the naming here, instead, as written in the commit message, it adds helper to generate supported formats of the PCM capability from the pipeline format (buffer allocated format).

`S32_LE,S24_LE,S16_LE' is the format we used to denote the supported format of PCM capability in our m4.

So here, it means for:
pcm specified with s16_le, the buffers are 16bit (2Bytes) size allocated, the PCM can support S16_LE at run-time only.
pcm specified with s24_le, the buffers are 24bit size allocated, the PCM can support both S24_LE and S16_LE at run-time.
pcm specified with s32_le, the buffers are 32bit size allocated, the PCM can support all S32_LE, S24_LE and S16_LE at run-time.

Reply:
can we rename the s32le names here, it's confusing with S32LE uppercase name as they are NOT the same thing.

Reply:
> can we rename the s32le names here, it's confusing with S32LE uppercase name as they are NOT the same thing.

They are existed for component format, and the uppercase names are for the PCM format. If want to change, we need another PR for to clean them up as there are hundreds of place where using them.